### DISCUSSION TITLE
[v3] $this->all() including model binding $post when updating

### DISCUSSION THREAD

**Discussion by addtobag** [livewire/livewire#5888] [1. Help]
When using $this->all() it includes $this->post, which leads to an error message as it's expecting post to be a field.

Error Message
"Add fillable property [post] to allow mass assignment on [App\Models\Post]."

```
public function setPost(Post $post)
    {
        $this->post = $post;
 
        $this->title = $post->title;
 
        $this->content = $post->content;
    }
 
    public function store()
    {
        Post::create($this->all());
    }
 
    public function update()
    {
        $this->post->update(
            $this->all()
        );
    }
```

**Comment by joshhanley** [comment]
@addtobag `->all()` returns all public properties, hence why you are seeing it. I would look at using a FormObect for your individual properties and calling `->all()` on the FormObject instead. See here https://livewire.laravel.com/docs/forms#extracting-a-form-object

Hope this helps!

**Reply by addtobag** [reply]
Hi Josh, 

Thanks for the reply, my example is from that documentation, and the example in the documentation the setPost method binds the model to $this->post which then is returned in ->all() which then causes the issue.

I of course can do the below, but i'm questioning if setting $this->post in the setPost method is wrong?

```
public function update()
    {
        $this->post->update(
            [
                'title' => $this->title,
                'content' => $this->content,
            ]
        );
    }

```

**Reply by joshhanley** [reply]
@addtobag can you link me to the part in the docs that has that? It probably needs changing.

**Reply by addtobag** [reply]
The very last example under "Extracting a form object":
https://livewire.laravel.com/docs/forms#extracting-a-form-object

Thanks

**Reply by krekas** [reply]
Having similar issue but with validation. When doing `$this->validate()` on the `CreatePost` component getting error `Cannot assign null to property App\Livewire\Forms\PostForm::$post of type App\Models\Post`

**Reply by addtobag** [reply]
Can you share your form object?

**Reply by krekas** [reply]
```php
class PostForm extends Form
{
    public Post $post;

    #[Rule('required|min:5')]
    public string $title = '';

    #[Rule('required|min:5')]
    public string $body = '';

    public function setPost(Post $post): void
    {
        $this->post = $post;

        $this->title = $post->title;

        $this->body = $post->body;
    }

    public function save(): void
    {
        Post::create([
            'title' => $this->title,
            'body' => $this->body,
        ]);
    }

    public function update(): void
    {
        $this->post->update([
            'title' => $this->title,
            'body' => $this->body,
        ]);
    }
}
```

**Reply by addtobag** [reply]
try 

`public ?Post $post = null;`

**Reply by krekas** [reply]
Oh right. Stupid mistake.

**Reply by joshhanley** [reply]
@addtobag thanks for helping out! 🙂

**Reply by krekas** [reply]
@joshhanley the docs should be updated accordingly I guess

**Reply by joshhanley** [reply]
@joshhanley yep agreed, PRs are welcome! 🙂
